### PR TITLE
adds a bookmark flag to MBArticle and a method to set this field in t…

### PR DESCRIPTION
**Pull request by jdwitten**
* Adds a bookmark flag to MBArticle and a method to set this field in the Store
* Also reuses MBArticleViewController for bookmarks

(read my TODO comment on the bookmark method)

### CODE REVIEW

File: iOS Client/ArticlesController/MBArticleDetailViewController.swift [hunk 1]
```diff
@@ -53,6 +58,23 @@ class MBArticleDetailViewController: UIViewController, WKNavigationDelegate {
         }
     }
     
+   @objc func bookmark(_ sender: Any) {
```
Comment (AlexRamey) [inline iOS Client/ArticlesController/MBArticleDetailViewController.swift:61]: is this @objc stuff necessary. Can't we do an IBAction? Idk about this, it's been a while

File: iOS Client/ArticlesController/MBArticleDetailViewController.swift [hunk 2]
```diff
@@ -53,6 +58,23 @@ class MBArticleDetailViewController: UIViewController, WKNavigationDelegate {
...
+        do {
+            guard let appDelegate = UIApplication.shared.delegate as? AppDelegate,
+                let container = appDelegate.persistentContainer,
+                let article = selectedArticle else {
+                return
+            }
+            
+            try MBStore().bookmark(article: article, persistentContainer: container)
+            // TODO: Dispatch an action here to update state if we change our models to be structs
```
Comment (AlexRamey) [inline iOS Client/ArticlesController/MBArticleDetailViewController.swift:70]: good point, I'll make this my highest priority after the image loading

File: iOS Client/BookmarksController/BookmarksCoordinator.swift [hunk 3]
```diff
@@ -25,7 +25,7 @@ class BookmarksCoordinator: NSObject, Coordinator, StoreSubscriber {
     
     // MARK: - Coordinator
     func start() {
-        let bookmarksController = MBBookmarksViewController.instantiateFromStoryboard()
+        let bookmarksController = MBArticlesViewController.instantiateFromStoryboard(bookmarks: true)
```
Comment (AlexRamey) [inline iOS Client/BookmarksController/BookmarksCoordinator.swift:28]: I don't know about re-using the same VC man. Let's discuss this tomorrow.

File: iOS Client/Store/MBStore.swift [hunk 4]
```diff
@@ -38,6 +38,19 @@ class MBStore: NSObject {
...
     
+    func getBookmarkedArticles(persistentContainer: NSPersistentContainer) -> [MBArticle]{
+        let fetchRequest = NSFetchRequest<NSManagedObject>(entityName: MBArticle.entityName)
+        let predicate = NSPredicate(format: "bookmarked == TRUE")
+        fetchRequest.predicate = predicate
+        return performFetch(managedContext: persistentContainer.viewContext, fetchRequest: fetchRequest) as? [MBArticle] ?? []
+
+    }
+    
+    func bookmark(article: MBArticle, persistentContainer: NSPersistentContainer) throws {
```
Comment (AlexRamey) [inline iOS Client/Store/MBStore.swift:49]: hmmm, this is assuming the MBArticle NSManagedObject is linked to the Persistent Container's View context. This happens to be the case, but it's a little dangerous to assume so.

I think you can just pass the article here, and use its managedObjectContext property to get the context, on which you can call save().

**Review by AlexRamey — COMMENTED** [review]
Left some comments for now, let's talk with DZ tomorrow about how this should look. I'm a little unsure about re-using the VC we have for articles home.

**Comment by jdwitten** [comment]
Closing this because we will implement bookmark with its own VC

**Pull request closed without merging**